### DISCUSSION TITLE
Interpolate custom properties with @apply

### DISCUSSION THREAD

**Discussion by kissu** [tailwindlabs/tailwindcss#1549] [General]
Sorry if it was already asked but I cannot find any answer or I'm maybe using the bad keywords.

I was wondering if we could use something like 
```css
@apply bg-(var(--color))-(var(--amount));
```
with that code from within a `.vue` component.
```vue
<div style="--color: green; --amount: 400;"></div>
```

and get the following result

```css
@apply bg-green-400;
```

Oh and keep `purgeCSS` working still. :hugs:

**Comment by adamwathan** [comment]
Impossible to make this work I'm afraid, `apply` calls are flattened into static CSS at build time so you can't change what they do at run-time with CSS variables.

**Reply by kissu** [reply]
Alright, just wanted to be sure. Thank you ! 🙏🏻
PS: I love your work, thanks again. 😍
